tests/tests_app/runners/test_cloud.py

```diff
@@ -702,3 +703,20 @@ def test_project_has_sufficient_credits():
     for balance, result in credits_and_test_value:
         project = V1Membership(name="test-project1", project_id="test-project-id1", balance=balance)
         assert cloud_runtime._project_has_sufficient_credits(project) is result
+
+
+def test_incompatible_cloud_compute_and_build_config():
```

Comment:
I tried to create a test that runs the CloudRuntime with relevant parts mocked, but failed. 
It would be great if we could refactor the huge dispatch method in logical chunks of function calls such that individual parts can be mocked/tested.

Reply:
Agreed, can you open an issue?

Reply:
Done. #15012